### PR TITLE
add binlog deploy and check process

### PR DESCRIPTION
**Pull request by shuijing198799**
1 add binlog deployment
2 add binlog check
3 change the main process
@weekface @xiaojingchen @zyguan @cwen0  PTAL

### CODE REVIEW

File: tests/actions.go [hunk 1]
```diff
@@ -245,8 +251,16 @@ func (oa *operatorActions) DeployTidbCluster(info *TidbClusterInfo) error {
...
+	initSecretName, backupSecretName, err := oa.CreateSecret(info)
+	if err != nil {
+		return fmt.Errorf("failed to create secret of cluster [%s]: %v", info.ClusterName, err)
+	}
+
+	info.InitSecretName, info.BackupSecretName = initSecretName, backupSecretName
+
 	cmd := fmt.Sprintf("helm install /charts/%s/tidb-cluster  --name %s --namespace %s --set-string %s",
-		info.OperatorTag, info.ClusterName, info.Namespace, info.HelmSetString())
+		info.OperatorTag, info.ClusterName, info.Namespace, info.HelmSetString(map[string]string{}))
```
Comment (weekface) [inline tests/actions.go]: map[string]string{}  -> nil

File: tests/actions.go [hunk 2]
```diff
@@ -92,7 +91,7 @@ type OperatorActions interface {
 	Restore(from *TidbClusterInfo, to *TidbClusterInfo) error
 	CheckRestore(from *TidbClusterInfo, to *TidbClusterInfo) error
 	ForceDeploy(info *TidbClusterInfo) error
-	CreateSecret(info *TidbClusterInfo) error
+	CreateSecret(info *TidbClusterInfo) (string, string, error)
```
Comment (weekface) [inline tests/actions.go]:
```
CreateInitSecret(info *TidbClusterInfo) (*corev1.Secret, error)
CreateBackupSecret(info *TidbClusterInfo) (*corev1.Secret, error)
```

I think it is better to create these secrets separately

File: tests/actions.go [hunk 3]
```diff
@@ -414,7 +428,7 @@ func chartPath(name string, tag string) string {
 
 func (oa *operatorActions) ScaleTidbCluster(info *TidbClusterInfo) error {
 	cmd := fmt.Sprintf("helm upgrade %s %s --set-string %s",
-		info.ClusterName, chartPath("tidb-cluster", info.OperatorTag), info.HelmSetString())
+		info.ClusterName, chartPath("tidb-cluster", info.OperatorTag), info.HelmSetString(map[string]string{}))
```
Comment (weekface) [inline tests/actions.go]: ditto

File: tests/actions.go [hunk 4]
```diff
@@ -425,7 +439,7 @@ func (oa *operatorActions) ScaleTidbCluster(info *TidbClusterInfo) error {
 
 func (oa *operatorActions) UpgradeTidbCluster(info *TidbClusterInfo) error {
 	cmd := fmt.Sprintf("helm upgrade %s %s --set-string %s",
-		info.ClusterName, chartPath("tidb-cluster", info.OperatorTag), info.HelmSetString())
+		info.ClusterName, chartPath("tidb-cluster", info.OperatorTag), info.HelmSetString(map[string]string{}))
```
Comment (weekface) [inline tests/actions.go]: ditto

File: tests/actions.go [hunk 5]
```diff
@@ -1458,13 +1448,19 @@ func (oa *operatorActions) getBackupDir(info *TidbClusterInfo) ([]string, error)
...
+	time.Sleep(20 * time.Second)
+
 	res, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput()
 	if err != nil {
-		glog.Errorf("cluster:[%s/%s] exec :%s failed,error:%v,result:%s", info.Namespace, info.ClusterName, cmd, err, res)
+		glog.Errorf("cluster:[%s/%s] exec :%s failed,error:%v,result:%s", info.Namespace, info.ClusterName, cmd, err, string(res))
 		return nil, err
 	}
 
+	if err != nil {
```
Comment (aylei) [inline tests/actions.go]: unreachable

Comment (shuijing198799) [reply tests/actions.go]: It seems you are right

File: tests/actions.go [hunk 6]
```diff
@@ -1458,13 +1448,19 @@ func (oa *operatorActions) getBackupDir(info *TidbClusterInfo) ([]string, error)
 	}
 
 	cmd := fmt.Sprintf("kubectl exec %s -n %s ls /data", getBackupDirPodName, info.Namespace)
-	glog.Infof(cmd)
+
+	time.Sleep(20 * time.Second)
```
Comment (aylei) [inline tests/actions.go]: Can we poll the following condition so we can proceed on earlier if condition meet?

Comment (shuijing198799) [reply tests/actions.go]: I try to poll at first but the time is hard to controller in one  minute. for every minute generate a cronjob to backup. At last we have to check the backup number.

File: tests/actions.go [hunk 7]
```diff
@@ -1475,9 +1475,182 @@ func (info *TidbClusterInfo) FullName() string {
 }
 
 func (oa *operatorActions) DeployIncrementalBackup(from *TidbClusterInfo, to *TidbClusterInfo) error {
+	glog.Infof("begin to deploy incremental backup cluster[%s] namespace[%s]", from.ClusterName, from.Namespace)
+	defer func() {
+		glog.Infof("deploy incremental backup end cluster[%s] namespace[%s]", to.ClusterName, to.Namespace)
+	}()
+	sets := map[string]string{
```
[the remark this answers is not in the document]
Comment (shuijing198799) [reply tests/actions.go]: done!

File: tests/actions.go [hunk 8]
```diff
@@ -1212,23 +1210,23 @@ func (info *TidbClusterInfo) QueryCount() (int, error) {
 	return 0, fmt.Errorf("can not find count of ")
 }
 
-func (oa *operatorActions) CreateSecret(info *TidbClusterInfo) error {
-	initSecretName := "set-secret"
-	backupSecretName := "backup-secret"
+func (oa *operatorActions) CreateSecret(info *TidbClusterInfo) (string, string, error) {
+	initSecretName := fmt.Sprintf("%s-set-secret", info.ClusterName)
```
Comment (xiaojingchen) [inline tests/actions.go]: the field should be set when `info` created

Comment (shuijing198799) [reply tests/actions.go]: done

**Pull request merged**